src/coreclr/jit/lsraarm64.cpp

```diff
@@ -25,6 +25,415 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 #include "sideeffects.h"
 #include "lower.h"
 
+//------------------------------------------------------------------------
+// getNextConsecutiveRefPosition: Get the next subsequent refPosition.
+//
+// Arguments:
+//    refPosition   - The refposition for which we need to find next refposition
+//
+// Return Value:
+//    The next refPosition or nullptr if there is not one.
+//
+RefPosition* LinearScan::getNextConsecutiveRefPosition(RefPosition* refPosition)
+{
+    assert(compiler->info.compNeedsConsecutiveRegisters);
+    RefPosition* nextRefPosition;
+    assert(refPosition->needsConsecutive);
+    nextConsecutiveRefPositionMap->Lookup(refPosition, &nextRefPosition);
+    assert((nextRefPosition == nullptr) || nextRefPosition->needsConsecutive);
+    return nextRefPosition;
+}
+
+//------------------------------------------------------------------------
+// setNextConsecutiveRegisterAssignment: For subsequent refPositions, set the register
```

Comment:
`setNext...` seems to imply it only assigns one. Would it be better named `AssignConsecutiveRegisters`?